sdk/core/azure-core/pom.xml

```diff
@@ -239,10 +250,52 @@
         </configuration>
       </plugin>
 
+      <plugin>
+        <groupId>org.jacoco</groupId>
+        <artifactId>jacoco-maven-plugin</artifactId>
+        <version>0.8.5</version> <!-- {x-version-update;org.jacoco:jacoco-maven-plugin;external_dependency} -->
+        <configuration>
+          <excludes>
+            <exclude>META-INF/**</exclude>
+          </excludes>
+        </configuration>
+      </plugin>
     </plugins>
   </build>
 
   <profiles>
+    <profile>
+      <id>java9-plus</id>
+      <activation>
+        <jdk>[9,)</jdk>
+      </activation>
+
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>3.8.1</version> <!-- {x-version-update;org.apache.maven.plugins:maven-compiler-plugin;external_dependency} -->
+            <executions>
+              <execution>
+                <id>java9-plus-mr</id>
+                <goals>
+                  <goal>compile</goal>
+                </goals>
+                <configuration>
+                  <release>9</release>
+                  <compileSourceRoots>
+                    <compileSourceRoot>${project.basedir}/src/main/java9</compileSourceRoot>
```

Comment:
Is the plan to eventually support Java8+ for all libraries? Is there a scenario where customers may want say Java9 but not Java11?

Reply:
Also, we have added it as a separate profile, will we have a pipeline that will build/test this profile on a regular cadence

Reply:
The current plan is to scope multi-release JAR functionality to a few classes in azure-core. 

All this is doing is compiling Java-version specific implementation into `/META-INF/versions/{Java version}`. Then when the JAR is used in an application and the class is being loaded, if Java 9 or later is being used it will look-up in `/META-INF/versions` for the Java version being used and attempt to find the class to load there, then it will cascade down the versions (11, 10, 9, etc) until it either finds the class or will default to the non-`/META-INF/versions` implementation. So, customers using Java 9, or Java 8 or any version without version specific implementation, will fallback to the baseline (Java 8) implementation.

Reply:
> Also, we have added it as a separate profile, will we have a pipeline that will build/test this profile on a regular cadence

Yes, this will be built and tested in the Java 11 CI runs

Reply:
>Is there a scenario where customers may want say Java9 but not Java11

Customers can pick any Java version equal to or above 8 and our libraries should work. With multi-release jar we are just making use of newer (better) language features if the customer picked a higher Java version. This should all be implementation details that the user doesn't need to be aware of.